src/io/flutter/run/daemon/FlutterDaemonController.java

```diff
@@ -116,7 +114,7 @@ void removeDeviceId(String deviceId) {
   public void forkProcess(Project project) throws ExecutionException {
     try {
       GeneralCommandLine commandLine = createCommandLine(project);
-      myHandler = new OSProcessHandler(commandLine);
+      myHandler = new DaemonProcessWrapper(new OSProcessHandler(commandLine));
```

Comment:
With this approach we'd have a single `flutter daemon` process for each app (i.e. flutter project directory). There's a single `FlutterDaemonController` in charge of each external daemon process. So yes, the controller can be shared between running apps provided they are started from the same directory but run on different devices. And the device polling job can be handled by the same process. That was part of the original design.

I didn't create a new subclass because Window and Linux/Mac have different subclasses of ProcessHandler already, so we would have needed to do it twice. The wrapper is intended to fix the cases where the debugger uses object identity to identify debugger tabs.

However, there are enough other odd things going on, and maintenance is likely to be so confusing, that I'm experimenting with a non-shared daemon approach. We'll end up with a lot of external process (potentially) but that's not really a big deal. Having readable code is probably more important than an elegant system architecture.